lib/js/jsbigints.nim

```diff
@@ -0,0 +1,229 @@
+
+func `xor`*(x, y: JsBigInt): JsBigInt {.importjs: "(# ^ #)".} =
+  runnableExamples:
+    doAssert (big"555" xor big"2") == big"553"
+
+func `shl`*(a, b: JsBigInt): JsBigInt {.importjs: "(# << #)".} =
+  runnableExamples:
+    doAssert (big"999" shl big"2") == big"3996"
+
+func `shr`*(a, b: JsBigInt): JsBigInt {.importjs: "(# >> #)".} =
+  runnableExamples:
+    doAssert (big"999" shr big"2") == big"249"
+
+func `-`*(a: JsBigInt): JsBigInt {.importjs: "($1#)".} =
+  runnableExamples:
+    doAssert -(big"10101010101") == big"-10101010101"
+
+func inc*(a: var JsBigInt; b: JsBigInt) {.importjs: "([#][0][0] += #)".} =
+  runnableExamples:
+    var big1: JsBigInt = big"1"
+    let big2: JsBigInt = big"2"
+    inc big1, big2
+    doAssert big1 == big"3"
+
+func dec*(a: var JsBigInt; b: JsBigInt) {.importjs: "([#][0][0] -= #)".} =
+  runnableExamples:
+    var big1: JsBigInt = big"1"
+    let big2: JsBigInt = big"2"
+    dec big1, big2
+    doAssert big1 == big"-1"
+
+func `+=`*(x: var JsBigInt; y: JsBigInt) {.importjs: "([#][0][0] $1 #)".} =
+  runnableExamples:
+    var big1: JsBigInt = big"1"
+    let big2: JsBigInt = big"2"
+    big1 += big2
+    doAssert big1 == big"3"
+
+func `-=`*(x: var JsBigInt; y: JsBigInt) {.importjs: "([#][0][0] $1 #)".} =
+  runnableExamples:
+    var big1: JsBigInt = big"1"
+    let big2: JsBigInt = big"2"
+    big1 -= big2
+    doAssert big1 == big"-1"
+
+func `*=`*(x: var JsBigInt; y: JsBigInt) {.importjs: "([#][0][0] $1 #)".} =
+  runnableExamples:
+    var big1: JsBigInt = big"2"
+    let big2: JsBigInt = big"4"
+    big1 *= big2
+    doAssert big1 == big"8"
+
+func inc*(x: var JsBigInt) {.importjs: "(++[#][0][0])".} =
+  runnableExamples:
+    var big1: JsBigInt = big"1"
+    inc big1
+    doAssert big1 == big"2"
+
+func dec*(x: var JsBigInt) {.importjs: "(--[#][0][0])".} =
+  runnableExamples:
+    var big1: JsBigInt = big"3"
+    dec big1
+    doAssert big1 == big"2"
+
+func `%=`*(x: var JsBigInt; y: JsBigInt) {.importjs: "([#][0][0] %= #)".} =
+  runnableExamples:
+    var big1: JsBigInt = big"10"
+    big1 %= big"2"
+    doAssert big1 == big"0"
+
+func `/=`*(x: var JsBigInt; y: JsBigInt) {.importjs: "([#][0][0] /= #)".} =
+  runnableExamples:
+    var big1: JsBigInt = big"10"
+    big1 /= big"2"
+    doAssert big1 == big"5"
+
+proc addInt*(result: var string; integer: JsBigInt) =
+  ## Converts `integer` to its string representation and appends it to `result`.
+  runnableExamples:
+    var a = "12345678"
+    let b = big"99999999999999999999999999999999999999"
+    a.addInt b
+    doAssert a == "1234567899999999999999999999999999999999999999"
+  result.add integer.toString
+
+proc `+`*(_: JsBigInt): JsBigInt {.error.} # Can not be used by design.
+  ## **Do NOT use.** https://github.com/tc39/proposal-bigint/blob/master/ADVANCED.md#dont-break-asmjs
+
+proc low*(_: JsBigInt): JsBigInt {.error.} ## **Do NOT use.**
+
+proc high*(_: JsBigInt): JsBigInt {.error.} ## **Do NOT use.**
```

Comment:
still not sure what's the point of this proc, but at least if you add it it should be:
```nim
proc high*(_: typedesc[JsBigInt]): JsBigInt {.error.} ## **Do NOT use.**
```
so that on use it shows: ` Error: usage of 'high' is an {.error.} defined at ` instead of a generic sigmatch error (try it with `let a = JsBigInt.high`)

ditto for `low`

Reply:
it has no `low` nor `high`, it depends on the interpreter, and interpreter depends on free memory at the moment of running the code.

Reply:
I understand it has no low or high, but my point was that omitting `proc low` and `proc high` would have same effect, you'd still have `doAssert not compiles(low(JsBigInt))`

Reply:
Yes, but worse error message "type mismatch".